src/line-graph.js

```diff
@@ -53,98 +31,143 @@ export class LineGraph extends HTMLElement {
+  }
 
   /**
-   * @name disconnectedCallback
-   * @description
-   * Call back for when the component is detached from the DOM
-   */
-  disconnectedCallback() { }
-
-  /**
-   * @name renderGraph
-   * @param {Array} data
-   * @param {innerHTML} el
-   * @description
-   * Renders the graph using d3js
+   * make series object
+   * @param {stringyfy data} data 
+   * @returns Array of objects
    */
-  renderGraph(data, el) {
-    // Setup the margins and height, width
-    var margin = JSON.parse(this.dataset.margin);
-    var height = parseInt(this.dataset.height);
-    var width = parseInt(this.dataset.width);
-    var unit = this.dataset.unit;
-
-    // Create X time scale
-    var xScale = d3.scaleTime()
-      .domain(d3.extent(data[0].datapoints, d => d.time))
-      .range([0, width - margin.bottom]);
+  series(data) {
+    let jsonObj = JSON.parse(data);
+    let grouping = this.parseData(jsonObj);
+    let series = [];
+    grouping.forEach(e => {
+      series.push(this.seriesData(e.group, e.datapoints));
+    })
 
-    // Create Y linear scale
-    var yScale = d3.scaleLinear()
-      .domain([0, d3.max(data[0].datapoints, d => d.value)])
-      .range([height - margin.left, 0]);
+    console.log("series ", series);
 
-    // Setup the svg element in the DOM
-    var svg = d3.select(el)
-      .style("width", width + margin.left + +margin.right)
-      .style("height", height + +margin.top + +margin.bottom)
-      .append('g')
-      .attr("transform", `translate(${margin.top}, ${margin.left})`);
-
-    // Create the lines
-    var line = d3.line()
-      .x(d => xScale(d.time))
-      .y(d => yScale(d.value));
-
-    // add element for line and add class name
-    let lines = svg.append('g')
-      .attr('class', 'lines');
-
-    // add the lines for each collection of objects to the SVG
-    lines.selectAll('.line-group')
-      .data(data).enter()
-      .append('g')
-      .attr('class', 'line-group')
-      .append('path')
-      .attr('class', 'line')
-      .attr('d', d => line(d.datapoints))
-      .style('stroke', this.dataset.lineColor)
-      .style('fill', 'none');
-    /*
-    TODO: Color schema strategy needed to ensure lines
-    are the right colors
-    https://github.com/d3/d3-scale/blob/master/README.md#sequential-scales
-    var color = d3.scaleOrdinal(d3.schemeCategory10);
-    .style('stroke', (d, i) => color(i));
-    */
+    return series;
+  }
 
-    // Configure X Axis ticks and add xScale
-    var xAxis = d3.axisBottom(xScale).ticks(5);
-    // Configure Y Axis ticks and
-    var yAxis = d3.axisLeft(yScale).ticks(5).tickFormat((d) => {
-      return new AxisLeft().convert(unit, d);
-    });
+  /**
+   * 
+   * @param {name for series} name 
+   * @param {datapoints with time and value} datapoints 
+   */
+  seriesData(name, datapoints) {
+    let arr = [];
+    datapoints.forEach(e => {
+      arr.push([new Date(e.time).getTime(), e.value])
+    })
+    return {
```

Comment:
hey, we can make a separate function and then we can assign the color

Reply:
yes.. Created function and add logic inside of that.

Reply:
its done !